src/openpgp/packet/signature.iced

```diff
@@ -257,14 +257,16 @@ class Signature extends Packet
         packets
 
       when T.direct
-        [ @primary].concat data_packets
+        [ @primary ].concat data_packets
 
       else
         err = new Error "cannot verify sigtype #{@type}"
         []
 
     # Now actually check that the signature worked.
-    unless err?
+    # We might want to skip verification of 3rd party signatures for
+    # certain packet types, like key_revocation (pgp designated revoke).
+    unless err? or (@is_third_party and @type is T.key_revocation)
```

Comment:
yes, it's good. I wonder if we should split this function up in to smaller people. And/or allow early return in the case of errors?  I had previously avoided that, but this function is getting more and more complex